activemodel/lib/active_model/type/registry.rb

```diff
@@ -9,15 +9,15 @@ def initialize
       end
 
       def register(type_name, klass = nil, **options, &block)
-        block ||= proc { |_, *args| klass.new(*args) }
+        block ||= proc { |_, *args, **kwargs| klass.new(*args, **kwargs) }
```

Comment:
We can't use `**kwargs` in this proc since the registered `klass.new` might not allow empty hash (e.g. `Integer.new`).
https://buildkite.com/rails/rails/builds/65803#d383432f-7419-4a27-91a9-6843aac234f1/934-942

How about to use `Proc#ruby2_keywords`?
That is provided to address this line.
https://bugs.ruby-lang.org/issues/16404

Reply:
Fixed.